pyttb/sptensor.py

```diff
@@ -860,32 +1232,56 @@ def mask(self, W: sptensor) -> np.ndarray:
 
     def mttkrp(self, U: Union[ttb.ktensor, List[np.ndarray]], n: int) -> np.ndarray:
         """
-        Matricized tensor times Khatri-Rao product for sparse tensor.
+        Matricized tensor times Khatri-Rao product for sparse tensor. This is
+        an efficient form of the matrix product that avoids explicitly
+        computing the matricized sparse tensor and the large intermediate
+        Khatri-Rao product arrays.
+
+        If the input includes a list of 2-D arrays (factor_matrices), this
+        computes a matrix product of the mode-`n` matricization of the sparse
+        tensor the with Khatri-Rao product of all arrays in the list except
+        the `n`th. The length of the list of arrays must equal the number of
+        dimensions of the sparse tensor. The shapes of each array must have
+        leading dimensions equal to the dimensions of the sparse tensor and
+        the same second dimension.
+
+        If the input is a :class:`pyttb.ktensor`, this
+        computes a matrix product of the mode-`n` matricization of the sparse
+        tensor the with Khatri-Rao product formed by the `factor_matrices` and
+        `weights` from the ktensor, excluding the `n`th factor matrix and
+        corresponding weight. The shape of the ktensor must be compatible with
+        the shape of the sparse tensor.
 
         Parameters
         ----------
         U:
-            Matrices to create the Khatri-Rao product
+            List of 2-D arrays or ktensor.
```

Comment:
Same conceptual note about our documentation adding contextual information beyond types. Since the types are already present and testable.